### ISSUE TITLE
sitewide, design system, CMS — 508-defect-2 [AXE-CORE]: Heading levels SHOULD increase by one, On this page component

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#7877]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

Team & project: vsa-public-websites, AMA follow-on

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

Heading levels **should** only increase by one. On FAQs — [https://preview-prod.vfs.va.gov/preview?nodeId=3004](https://preview-prod.vfs.va.gov/preview?nodeId=3004), the On this page component currently uses an H3 although it generally follows the H1, and there's no H2 preceding it. 

### Why it Matters
The underlying purpose of headers is to convey the structure of the page. For sighted users, the same purpose is achieved using different sizes of text. Text size, however, is not helpful for users of screen readers, because a screen reader identifies a header only if it is properly marked-up. When heading elements are applied correctly, the page becomes much easier to navigate for screen reader users and sighted users alike.

In the same way that sighted users can glance at a page and get a sense of its contents, users of screen readers can do the same by navigating through headings. Well written and properly ordered headings can save users, especially those who use screen readers, a lot of time and frustration.

The purpose of headings is to describe the structure of the webpage, not just highlight important text. They should be brief, clear, unique, and marked with h1 through h6 elements applied in hierarchical order. All of these qualities make headings valuable tools for screen reader users. Similar to the way sighted users can glance at a page and get a sense of its contents, screen reader users can navigate through headings. Well written and properly ordered headings can save screen reader time and frustration.

In addition to making the page more accessible, headings have other benefits since search engines use headings when filtering, ordering, and displaying results. Improving the accessibility of your site can also have the effect of making your page more findable.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

- As a screen reader user, I want to navigate the hierarchy of the page content using heading levels to save time and frustration.
- As a screen reader user, I want to understand when there is a table of contents I may use to more easily navigate the page content.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: any
* Server destination: staging & production

## Steps to Recreate

1. Enter https://preview-prod.vfs.va.gov/preview?nodeId=3004 in browser, in the VA environment
1. Have developer tools open, and the axe browser extension loaded
1. Verify that the heading levels error displays
1. Optionally, run a screen reader and navigate by headings. Verify that this component is communicated as an H3 without a preceding H2. If it is changed to an H2, verify that it is communicated to be the same as an H2 in the content, which semantically isn't the same. For non-sighted users this may be unclear.

### Current code

`<h3 id="on-this-page">`

### Potential fixes

**Preferred solution** 

    ```html
        <nav aria-label="table of contents">
          <dl>
            <dt role="heading" class="vads-u-font-size--h3">On this page</dt>
            <dd><i class="fas fa-arrow-down" aria-hidden="true" role="presentation"></i><a href="#">link</a></dd>
            <dd><i class="fas fa-arrow-down" aria-hidden="true" role="presentation"></i><a href="#">link</a></dd>
            <dd><i class="fas fa-arrow-down" aria-hidden="true" role="presentation"></i><a href="#">link</a></dd>
          </dl>
        </nav>
    ```

## WCAG or Vendor Guidance (optional)

* [axe-core 3.4 - Heading levels should only increase by one](https://dequeuniversity.com/rules/axe/3.4/heading-order)
* [MDN dialog element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog)
* [MDN ARIA: alert role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Alert_Role)
* [MDN Description List element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl)

**Comment by joshkimux** [comment]
This will be addressed in [#21957](https://github.com/department-of-veterans-affairs/va.gov-team/issues/21957), so I'll close this out!

@1Copenut for awareness
